File: Capítulo_5/Deque/Removendo elementos no final do deque/removeBack.js
// removeBack(): esse método remove o último elemento do deque.
class Deque {
    constructor() {
        this.count = 0;
        this.lowestCount = 0;
        this.items = {};
    }

    removeBack() {
        if (this.isEmpty()) {
            return undefined;
        }
        this.count--;
        const result = this.items[this.count];
        delete this.items[this.count];
        return result;        
    }
}

File: Capítulo_5/Deque/Removendo elementos no final do deque/Código em Python/removeBack.py
# removeBack(): esse método remove o último elemento do deque.
class Deque:
    def __init__(self):
        self.count = 0
        self.lowest_count = 0
        self.items = {}

    def removeBack(self):
        if (self.is_empty()):
            return None
        self.count -= 1
        result = self.items[self.count]
        del self.items[self.count]
        return result
    